src/ImageSharp/Common/Helpers/RuntimeEnvironment.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace SixLabors.ImageSharp
+{
+    /// <summary>
+    /// Provides information about the .NET runtime installation.
+    /// Many methods defer to <see cref="RuntimeInformation"/> when available.
+    /// </summary>
+    internal static class RuntimeEnvironment
+    {
+        private static readonly Lazy<bool> IsNetCoreLazy = new Lazy<bool>(() => FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase));
+
+        /// <summary>
+        /// Gets a value indicating whether the .NET installation is .NET Core
+        /// </summary>
+        public static bool IsNetCore => IsNetCoreLazy.Value;
+
+        /// <summary>
+        /// Gets the name of the .NET installation on which an app is running.
+        /// </summary>
+        public static string FrameworkDescription => RuntimeInformation.FrameworkDescription;
```

Comment:
Does this have to be exposed? I would make it private.

Reply:
I did make a mistake yeah but corrected it in a later commit. 

What I've tried so far:

- Filtered out the runtime instrinsics part of the `UnPremultiply` method to aviod Mac on Core 3.1
- Rewrote the scalar overhang portion of the method to use a simple for loop over the span.

Neither worked. The only thing I could get working was duplicating the code which is simply awful. I couldn't replicate the issue locally either to effectively debug.

Regarding the method, no it doesn't need to be exposed.